app/scripts/modules/core/pipeline/config/stages/executionWindows/ExecutionWindowDayPickerComponent.js

```diff
@@ -0,0 +1,89 @@
+import angular from 'angular';
+
+class ExecutionWindowDayPickerController {
+
+  constructor() {
+    this.DAYS = [{
+      dow: 'sunday',
+      abbr: 'Sun'
+    }, {
+      dow: 'monday',
+      abbr: 'Mon'
+    }, {
+      dow: 'tuesday',
+      abbr: 'Tue'
+    }, {
+      dow: 'wednesday',
+      abbr: 'Wed'
+    }, {
+      dow: 'thursday',
+      abbr: 'Thu'
+    }, {
+      dow: 'friday',
+      abbr: 'Fri'
+    }, {
+      dow: 'saturday',
+      abbr: 'Sat'
+    }];
+    this.dayModel = {};
+  }
+
+  $onInit() {
+    console.log('executing onInit');
+    const days = new Set(this.days);
+    this.DAYS.forEach((day, index) => this.dayModel[day.dow] = days.has(index));
+  }
+
+  all() {
+    this.dayModel.sunday = this.dayModel.saturday = this.dayModel.monday =
+      this.dayModel.tuesday = this.dayModel.wednesday =
+        this.dayModel.thursday = this.dayModel.friday = true;
+    this.updateModel();
+  }
+
+  none() {
+    this.dayModel.sunday = this.dayModel.saturday = this.dayModel.monday =
+      this.dayModel.tuesday = this.dayModel.wednesday =
+        this.dayModel.thursday = this.dayModel.friday = false;
+    this.updateModel();
+  }
+
+  weekdays() {
+    this.dayModel.sunday = this.dayModel.saturday = false;
+    this.dayModel.monday = this.dayModel.tuesday = this.dayModel.wednesday =
+      this.dayModel.thursday = this.dayModel.friday = true;
+    this.updateModel();
+  }
+
+  weekend() {
+    this.dayModel.sunday = this.dayModel.saturday = true;
+    this.dayModel.monday = this.dayModel.tuesday = this.dayModel.wednesday =
+      this.dayModel.thursday = this.dayModel.friday = false;
+    this.updateModel();
+  }
+
+  updateModel() {
+    const updated = [];
+    this.DAYS.forEach((day, index) => {
+      if (this.dayModel[day.dow]) {
+        updated.push(index + 1);
+      }
+    });
+    this.days = updated;
+  }
+}
+
+const EXECUTION_WINDOW_DAY_PICKER_COMPONENT = {
+  bindings: {
+    days: '='
+  },
+  controller: ExecutionWindowDayPickerController,
+  controllerAs: 'dayPicker',
+  templateUrl: require('./ExecutionWindowDayPickerComponent.html')
+};
+
+const MODULE_NAME = 'spinnaker.core.pipeline.stage.executionWindows.dayPicker';
+angular.module(MODULE_NAME, [])
+  .component('executionWindowDayPicker', EXECUTION_WINDOW_DAY_PICKER_COMPONENT);
+export default MODULE_NAME;
```

Comment:
To be consistent with the rest of the code base, it's probably better to rewrite this block (L76-88) as:
```javascript
module.exports = angular.module('spinnaker.core.pipeline.stage.executionWindows.dayPicker', [])
  .component('executionWindowDayPicker', {
    bindings: {
      days: '='
    },
    controller: ExecutionWindowDayController,
    templateUrl: require('./executionWindowDayPicker.component.html')
});
```